src/transformers/models/fast/modeling_fast.py

```diff
@@ -0,0 +1,932 @@
+            config.backbone_stage4_groups,
+        ):
+            stage4.append(RepConvLayer(*stage_config))
+        self.stage4 = nn.ModuleList(stage4)
+
+    def forward(self, hidden_states):
+        hidden_states = self.first_conv(hidden_states)
+        output = []
+
+        for block in self.stage1:
+            hidden_states = block(hidden_states)
+        output.append(hidden_states)
+
+        for block in self.stage2:
+            hidden_states = block(hidden_states)
+        output.append(hidden_states)
+
+        for block in self.stage3:
+            hidden_states = block(hidden_states)
+        output.append(hidden_states)
+
+        for block in self.stage4:
+            hidden_states = block(hidden_states)
+        output.append(hidden_states)
+
+        return output
+
+
+class FASTNeck(nn.Module):
+    def __init__(self, config):
+        super().__init__()
+        reduce_layer_configs = list(
+            zip(
+                config.neck_in_channels,
+                config.neck_out_channels,
+                config.neck_kernel_size,
+                config.neck_stride,
+                config.neck_dilation,
+                config.neck_groups,
+            )
+        )
+        self.num_layers = len(reduce_layer_configs)
+        for layer_ix in range(0, len(reduce_layer_configs)):
+            setattr(self, f"reduce_layer{layer_ix + 1}", RepConvLayer(*reduce_layer_configs[layer_ix]))
+
+        self._initialize_weights()
+
+    def _initialize_weights(self):
+        for m in self.modules():
+            if isinstance(m, nn.Conv2d):
+                nn.init.kaiming_normal_(m.weight)
+            elif isinstance(m, nn.BatchNorm2d):
+                m.weight.data.fill_(1)
+                m.bias.data.zero_()
+
+    def _upsample(self, layer_out, height, width):
+        return F.upsample(layer_out, size=(height, width), mode="bilinear")
+
+    def forward(self, hidden_states):
+        first_layer_hidden = hidden_states[0]
+        first_layer_hidden = self.reduce_layer1(first_layer_hidden)
+        output_stages = [first_layer_hidden]
+
+        for layer_ix in range(1, self.num_layers):
+            layer_out = getattr(self, f"reduce_layer{layer_ix + 1}")(hidden_states[layer_ix])
+            layer_out = self._upsample(layer_out, first_layer_hidden[2], first_layer_hidden[3])
+            output_stages.append(layer_out)
+
+        combined_hidden_states = torch.cat(output_stages, 1)
+        return combined_hidden_states
+
+
+class FASTHead(nn.Module):
+    def __init__(self, config):
+        super().__init__()
+        self.conv = RepConvLayer(
+            config.head_conv_in_channels,
+            config.head_conv_out_channels,
+            config.head_conv_kernel_size,
+            config.head_conv_stride,
+            config.head_conv_dilation,
+            config.head_conv_groups,
+        )
+
+        self.final = ConvLayer(
+            config.head_final_in_channels,
```

Comment:
should we really make all of those attributes configurable? Might be fine for me to hardcode

Reply:
Stride varies for FASTConvLayer used in fast model , But dilation and groups can be removed